[STORYBOOK] 스토리북 생성 및 사칙연산 만들기

JH-design opened the pull request:
- 스토리북 다시 정리하여 새로 생성
- 사칙연산을 클래스 컴포넌트의 ref 이용하여 DOM 요소에 접근하여 화면에 나올 수 있게 함

JH-design committed:
[STORYBOOK] 스토리북 생성 및 사칙연산 만들기

- 스토리북 다시 정리하여 새로 생성
 - 사칙연산을 클래스 컴포넌트의 ref 이용하여 DOM 요소에 접근하여 화면에 나올 수 있게 함

gunwooCho commented:
내용 확인했습니다만 숫자 입력에대한 처리가 미흡합니다.

"정수" 만 받을 수 있고 표시도 정수로 표시할 수 있도록 수정해보세요

JH-design committed:
[STORYBOOK] 스토리북 생성 및 사칙연산 만들기

- 사칙연산을 정수로만 받아서 사용할 수 있도록 수정함
(지수 표기법이 나오지 않게 toString 메서드를 사용했으나 숫자가 클 경우 지수 표기법이 나오는 것을 확인)

gunwooCho commented:
12-과제1-두-숫자의-사칙연산의-결과를-출력하는-스토리를-작성하시오 브랜치에 pullRequest요청하도록 수정바랍니다.

gunwooCho commented on components/jhyu/2_Calculation.jsx:
```diff
@@ -1,36 +1,63 @@
-    this.result = React.createRef();
+    this.state = {
+      num1: '',
+      num2: '',
```
저장할떄 숫자로 저장하는게 어떨까요

gunwooCho commented on components/jhyu/2_Calculation.jsx:
```diff
@@ -1,36 +1,63 @@
+          onChange={event => this.setState({ num2: event.target.value.replace(/[^0-9]/g, '') })}
+        />
+        <button onClick={this.handleCalculate}>계산하기</button>
+        <pre>{result}</pre>
```
결과 출력에 pre로 작성한 이유가 있을까요

gunwooCho commented on components/jhyu/2_Calculation.jsx:
```diff
@@ -1,36 +1,63 @@
+        <input
+          type="text"
+          value={num1}
+          onChange={event => this.setState({ num1: event.target.value.replace(/[^0-9]/g, '') })}
```
이 작성법은 입력은 숫자 외에 입력이 가능하고 그 다음에 특정 문자를 고쳐주는 개념으로 보입니다.

이 내용으로 사용 시 입력은 가능한 현상이 있는 걸로 알고 있는데 수정 가능할까요

JH-design committed:
[STORYBOOK] 스토리북 생성 및 사칙연산 만들기

- 사칙연산을 정수로만 받아서 사용할 수 있도록 수정함
- 양의 정수, 음의 정수 모두 표현할 수 있도록 수정함
- 버튼쪽에 경고로 수정요청이 떠서,  type 추가함

JH-design committed:
[STORYBOOK] 스토리북 생성 및 사칙연산 만들기

- 사칙연산을 정수로만 받아서 사용할 수 있도록 수정함
- (-)는 한번만 나오도록 처리, (.) 아예 안나오게 처리
 -> 숫자 뒤에 (-) 나오는 이슈로 추후 수정 예정임 ㅜ

JH-design committed:
[STORYBOOK] 스토리북 생성 및 사칙연산 만들기_음수 수정

- (-)는 한번만 나오도록 처리, (.) 아예 안나오게 처리

gunwooCho commented on components/jhyu/2_Calculation.jsx:
```diff
@@ -7,42 +7,50 @@ class CalculationApp extends React.Component {
-    const product = (parsedNum1 * parsedNum2).toString();
-    const quotient = (parsedNum1 / parsedNum2).toString();
+    // 음수 표시 처리
+    const firstNum = isNegative ? parsedNum1 : parsedNum2;
```
여기 때문에 음수 결과가 정상적으로 표시되던가요?

gunwooCho reviewed (commented):
답변바랍니다.
